lib/ex_wubook/pci_proxies/channex_pci.ex

```diff
@@ -0,0 +1,177 @@
+    url = get_token_url(api_key)
+    body = get_token_body_request(user, password, provider_key)
+
+    with {:ok, body, _} <- send_request(url, body),
+         {:ok, %{param: [_, token]}} <- XMLRPC.decode(body) do
+      {:ok, token}
+    end
+  end
+
+  defp get_token_url(api_key), do: get_url(@targetURI, api_key)
+  defp get_card_token_url(api_key), do: get_url("#{@targetURI}&savecvv=true", api_key)
+
+  defp get_url(url, api_key) do
+    "#{pci_url()}?apikey=#{api_key}&method=post&profile=wubook&url=#{url}"
+  end
+
+  defp get_card_token(token, lcode, booking_code, password, api_key) do
+    url = get_card_token_url(api_key)
+    body = get_card_token_body(token, lcode, booking_code, password)
+
+    with {:ok, body, headers} <- send_request(url, body),
+         {:ok, %{param: [0, card_data]}} <- XMLRPC.decode(body) do
+      {
+        :ok,
+        %{
+          cardholder_name: card_data["cc_owner"],
+          card_number: card_data["cc_number"],
+          expiration_date: NormalizeExpirationDate.execute(card_data["cc_expiring"]),
+          card_type: @card_types[card_data["cc_type"]],
+          cvv: card_data["cc_cvv"],
+          token: headers[:token]
+        }
+      }
+    else
+      {:ok, %XMLRPC.MethodResponse{param: [-17, "No CC for this reservation"]}} ->
+        {:error, :no_cc_for_this_reservation}
+
+      {:ok, %XMLRPC.MethodResponse{param: [-100, "Invalid Input"]}} ->
+        {:error, :invalid_input}
+
+      {_, error} ->
+        {:error, error}
+
+      error ->
+        error
+    end
+  end
+
+  defp get_token_body_request(user, password, provider_key) do
+    """
+    <?xml version="1.0"?>
+    <methodCall>
+      <methodName>acquire_token</methodName>
+      <params>
+        <param>
+          <value>
+            <string>#{user}</string>
+          </value>
+        </param>
+        <param>
+          <value>
+            <string>#{password}</string>
+          </value>
+        </param>
+        <param>
+          <value>
+            <string>#{provider_key}</string>
+          </value>
+        </param>
+      </params>
+    </methodCall>
+    """
+  end
+
+  defp get_card_token_body(token, lcode, booking_code, password) do
+    """
+    <?xml version="1.0"?>
+    <methodCall>
+      <methodName>fetch_ccard</methodName>
+      <params>
+        <param><value><string>#{token}</string></value></param>
+        <param><value><int>#{lcode}</int></value></param>
+        <param><value><int>#{booking_code}</int></value></param>
+        <param><value><string>#{password}</string></value></param>
+      </params>
+    </methodCall>
+    """
+  end
+
+  defp send_request(url, body) do
+    with {:ok, response} <- request(:post, url, body, [], []) do
+      code = response.status
+
+      meta =
+        %{}
+        |> add_cards_info(response.headers)
+        |> finalize_meta(%{response: response.body, status_code: code})
+
+      cond do
+        code == 200 -> {:ok, response.body, meta}
```

Comment:
what about code > 200 ?

Reply:
It's copy -paste from pci_booking.ex now it's work, I don't want change it.

Reply:
By spec, they not use any other success code except 200